core/src/create.jl

```diff
@@ -624,33 +625,98 @@ end
+    active = BitVector()
+    min_level = Float64[]
+    return_factor = Float64[]
+    interpolations = Vector{ScalarInterpolation}[]
+
+    errors = false
+    trivial_timespan = [nextfloat(-Inf), prevfloat(Inf)]
+    t_end = seconds_since(config.endtime, config.starttime)
+
+    # Create a dictionary priority => time data for that priority
+    time_priority_dict = Dict(
+        first(group).priority => StructVector(group) for
+        group in IterTools.groupby(row -> row.priority, time)
     )
+
+    for node_id in node_ids
+        first_row = nothing
+        demand = Vector{ScalarInterpolation}()
+
+        if node_id in static_node_ids
+            rows = searchsorted(static.node_id, node_id)
+            static_id = view(static, rows)
+            for p in 1:maximum_priority
+                idx = findsorted(static_id.priority, p)
+                demand_p = !isnothing(idx) ? static_id[idx].demand : 0.0
+                demand_p_itp = LinearInterpolation([demand_p, demand_p], trivial_timespan)
+                push!(demand, demand_p_itp)
+            end
+            push!(interpolations, demand)
+            first_row = first(static_id)
+            is_active = coalesce(first_row.active, true)
+
+        elseif node_id in time_node_ids
+            for p in 1:maximum_priority
+                if p in keys(time_priority_dict)
+                    demand_p_itp, is_valid = get_scalar_interpolation(
+                        config.starttime,
+                        t_end,
+                        time_priority_dict[p],
+                        node_id,
+                        :demand;
+                        default_value = 0.0,
+                    )
+                    if is_valid
+                        push!(demand, demand_p_itp)
+                    else
+                        @error "The demand(t) relationship for User #$node_id of priority $p from the time table has repeated timestamps, this can not be interpolated."
+                        errors = true
+                    end
+                else
+                    demand_p_itp = LinearInterpolation([0.0, 0.0], trivial_timespan)
+                    push!(demand, demand_p_itp)
+                end
+            end
+            push!(interpolations, demand)
+
+            first_row_idx = searchsortedfirst(time.node_id, node_id)
+            first_row = time[first_row_idx]
+            is_active = true
+        else
+            @error "User node #$node_id data not in any table."
+            errors = true
+        end
+
+        if !isnothing(first_row)
+            min_level_ = coalesce(first_row.min_level, 0.0)
+            return_factor_ = first_row.return_factor
+            push!(active, is_active)
+            push!(min_level, min_level_)
+            push!(return_factor, return_factor_)
+        end
+    end
+
+    if errors
+        error("Errors occurred when parsing User data.")
+    end
+
+    allocated = [zeros(maximum_priority) for id in node_ids]
```

Comment:
Same issue here for large maximum_priority. Unless we map input of [4,8,1000] to [1,2,3] first.